letsencrypt/plugins/common.py

```diff
@@ -110,8 +110,17 @@ def __init__(self, tup):
     @classmethod
     def fromstring(cls, str_addr):
         """Initialize Addr from string."""
-        tup = str_addr.partition(':')
-        return cls((tup[0], tup[2]))
+        if str_addr.startswith('['):
+            # ipv6 addresses starts with [
+            endIndex = str_addr.rfind(']')
+            host = str_addr[:endIndex + 1]
+            port = ''
+            if len(str_addr) > endIndex + 3 and str_addr[endIndex + 2] == ':':
```

Comment:
str_addr[endIndex] + 2] should never be ":", if the port is set, that'll be the first digit of the port nr.